ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientFactory.java

```diff
@@ -32,8 +32,8 @@ public static QueuedIngestClient createClient(ConnectionStringBuilder csb) throw
      * @return a new queued ingest client
      * @throws URISyntaxException if the connection string is invalid
      */
-    public static QueuedIngestClient createClient(ConnectionStringBuilder csb, @Nullable HttpClientProperties properties) throws URISyntaxException {
-        return new QueuedIngestClientImpl(csb, properties);
+    public static QueuedIngestClient createClient(ConnectionStringBuilder csb, @Nullable HttpClientProperties properties, boolean autoCorrectEndpoint) throws URISyntaxException {
```

Comment:
These are all breaking changes, I'm afraid you'll have to do more overloads (cause java doesn't have default params)